financial-connections/src/main/java/com/stripe/android/financialconnections/repository/FinancialConnectionsAccountsRepository.kt

```diff
@@ -77,21 +84,17 @@ private class FinancialConnectionsAccountsRepositoryImpl(
 ) : FinancialConnectionsAccountsRepository {
 
     /**
-     * Ensures that manifest accesses via [getOrFetchAccounts] suspend until
+     * Ensures that [cachedAccounts] accesses via [getCachedAccounts] suspend until
      * current writes are running.
      */
     val mutex = Mutex()
-    private var cachedAccounts: PartnerAccountsList? = null
+    private var cachedAccounts: List<PartnerAccount>? = null
 
-    override suspend fun getOrFetchAccounts(
-        clientSecret: String,
-        sessionId: String
-    ): PartnerAccountsList =
-        mutex.withLock {
-            cachedAccounts ?: run {
-                postAuthorizationSessionAccounts(clientSecret, sessionId)
-            }
-        }
```

Comment:
We don't want to fetch accounts when they're not cached anymore. There could be different account fetch sources with networking, and the moment they're retrieved they're required to be there. Otherwise we'll throw an error.